keps/sig-node/4112-passdown-resources-to-cri/README.md

```diff
@@ -0,0 +1,1277 @@
+can be avoided by draining the node before updating the kubelet and the CRI
+runtime.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes, disabling the `KubeletContainerResourcesInPodSandbox` feature gate will
+disable the feature. Restarting pods may be needed to reset the information
+that was passed down to the CRI.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+New pods will have the feature enabled. Existing pods will continue to operate
+as before until restarted.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Unit tests for the feature gate will be added.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+Rollback or rollout in the kubelet should not fail - it only enables/disabled
+the information (fields in the CRI message) passed down to the CRI runtime.
+
+However, if the CRI runtime depends on the feature, a rollout or rollback may
+cause failures of applications on pod restarts. Running pods are not affected.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Alpha: No new metrics are planned. Non-ready pods with CreatePodSandboxError
```

Comment:
Are there any metrics around the GRPC calls from kubelet to CRI that could show errors?

Reply:
Good point. The exising `kubelet_started_pods_errors_total` metric == CreatePodSandboxErrors. I modified this section to

```
Alpha: No new metrics are planned. Increase in the existing
`kubelet_started_pods_errors_total` metric can indicate a problem caused by
this feature.

Generally, non-ready pods with CreatePodSandboxError status (reflected by the
`kubelet_started_pods_errors_total` metric) is a possible indicator. The error
message will contain details if the CRI failure is related to the feature.
```

@johnbelamaric PTAL

Reply:
Closing as resolved